core/ZigWrapperWriter.cpp

```diff
@@ -0,0 +1,366 @@
+                    }
+                    --m_indentationDepth;
+                    out << "}\n\n";
+
+                    // Now we'll write the bindings to the actual class methods. The name pattern here should hopefully not
+                    // ever cause conflicts with user defined symbols; I don't see any reasonable case where it would cause a
+                    // problem; any naming collisions will probably be a result of abuse rather than accidentally breaking
+                    // things.
+                    // out << std::string(m_indentationDepth, '\t') << "extern {\n";
+                    // ++m_indentationDepth;
+                    for (const auto &method : classNode->methods)
+                    {
+                        // out << std::format("\t"
+                        //                    R"(#[link_name = "{}"] fn polyglot_{}_method_{}(this: {})",
+                        //                    method.mangledName,
+                        //                    classNode->name,
+                        //                    method.functionName,
+                        //                    classNode->name);
+                        out << std::format("\t"
+                                           R"(fn polyglot_{}_method_{}(this: {})",
+                                           method.mangledName,
+                                           classNode->name,
+                                           method.functionName,
+                                           classNode->name);
+                        out << std::format("\t"
+                                           R"(@export(polyglot_{}_method_{}, {}))",
+                                           method.mangledName,
+                                           classNode->name,
+                                           method.functionName,
+                                           classNode->name);
+
+                        std::string params;
+                        for (const auto &param : method.parameters)
+                            params += ", " + param.name + ": " + getTypeString(param.type);
+                        out << params << ')';
+
+                        if (method.returnType.baseType != Type::Void)
+                            out << getTypeString(method.returnType);
+                        out << ";\n";
+                    }
+                    // --m_indentationDepth;
+                    // out << "}\n";
+                }
+            }
+        }
+
+        previousNodeType = node->nodeType();
+    }
+
+    if (previousNodeType == ASTNodeType::Function)
+        out << "}\n";
+
+    out.flush();
+
+    --s_onlyWriteHeaderOnce;
+}
+
+std::string ZigWrapperWriter::getTypeString(const QualifiedType &type) const
+{
+    std::string typeString;
+    if (type.isConst)
+        typeString += "const ";
+    if (type.isReference)
+        throw std::runtime_error("Zig not have reference keywords");
```

Comment:
It might be possible to create wrapper functions to convert references into pointers. That would probably be a subject for a future PR though.

Reply:
Exactly!

**Note:** The compatibility between `char*|const char*` and `[*:0]u8 or ?*u8|[*:0]const u8` has been tested until now. However, C++ std::string has never been tried before.